sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.controller.submitApplication.e2e-spec.ts

```diff
@@ -0,0 +1,358 @@
+            studyEndDate: secondApplicationOfferingInitialValues.studyEndDate,
+          },
+        },
+      ),
+    );
+    const secondApplicationProgram = secondApplicationOffering.educationProgram;
+    const applicationData = {
+      programYear,
+      selectedOfferingDate:
+        secondApplicationOfferingInitialValues.studyStartDate,
+      selectedOfferingEndDate:
+        secondApplicationOfferingInitialValues.studyEndDate,
+      programYearStartDate: programYear.startDate,
+      programYearEndDate: programYear.endDate,
+      howWillYouBeAttendingTheProgram:
+        secondApplicationOfferingInitialValues.offeringIntensity,
+      selectedProgram: secondApplicationProgram.id,
+      selectedOffering: secondApplicationOffering.id,
+    };
+    const payload = {
+      associatedFiles: [],
+      data: applicationData,
+      programYearId: programYear.id,
+    } as SaveApplicationAPIInDTO;
+    const endpoint = `/students/application/${secondDraftApplication.id}/submit`;
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+    const dryRunSubmissionMock = jest.fn().mockResolvedValue({
+      valid: true,
+      formName: FormNames.Application,
+      data: { data: applicationData },
+    });
+    formService.dryRunSubmission = dryRunSubmissionMock;
+    // Act/Assert
+    await request(app.getHttpServer())
+      .patch(endpoint)
+      .send(payload)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.UNPROCESSABLE_ENTITY)
+      .expect({
+        message:
+          "There is an existing application already with overlapping study period or a pending PIR.",
+        errorType: "STUDY_DATE_OVERLAP_ERROR",
+      });
+  });
+
+  it("Should throw study dates overlap error when an application submitted for a student via the SFAS system has overlapping study start or study end dates with another application.", async () => {
+    // Arrange
+    const sfasIndividual = await saveFakeSFASIndividual(db.dataSource, {
+      initialValues: {
+        lastName: sharedStudent.user.lastName,
+        birthDate: sharedStudent.birthDate,
+        sin: sharedStudent.sinValidation.sin,
+      },
+    });
+    const firstFakeSFASPartTimeApplication = createFakeSFASPartTimeApplication(
+      { individual: sfasIndividual },
+      {
+        initialValues: {
+          startDate: getISODateOnlyString(addDays(9990)),
+          endDate: getISODateOnlyString(addDays(10010)),
+        },
+      },
+    );
+    await db.sfasPartTimeApplications.save(firstFakeSFASPartTimeApplication);
+    // Create a fake offering for the second application.
+    const auditUser = await db.user.save(createFakeUser());
+    const secondApplicationOfferingInitialValues = {
+      studyStartDate: getISODateOnlyString(addDays(10000)),
+      studyEndDate: getISODateOnlyString(addDays(10020)),
+      offeringIntensity: OfferingIntensity.partTime,
+    };
```

Comment:
IMO for every test it is not required to create 2 offerings. If we re-use the same offering from application1 the overlap error will be thrown.

Reply:
Just want to check how do you want me to approach this. Since the offering dates of one test shouldn't conflict with the other test, I created separate dates for each test.